ibis/expr/schema.py

```diff
@@ -182,3 +185,78 @@ def schema_from_pairs(lst):
 @schema.register(collections.abc.Iterable, collections.abc.Iterable)
 def schema_from_names_types(names, types):
     return Schema(names, types)
+
+
+def coerce_to_dataframe(
+    data: Any, column_names: List[str], types: List[dt.DataType]
+) -> pd.DataFrame:
+    """Coerce the following shapes to a DataFrame.
+
+    The following shapes are allowed:
+    (1) A list/tuple of Series
+    (2) A list/tuple np.ndarray
+    (3) A list/tuple of scalars
+    (4) A Series of list/tuple
+    (5) pd.DataFrame
+
+    Note:
+    This method does NOT always return a new DataFrame. If a DataFrame is
+    passed in, this method will return the original object.
+
+    Parameters
+    ----------
+    data : pd.DataFrame, a tuple/list of pd.Series, a tuple/list of np.array
+    or a pd.Series of tuple/list
+
+    column_names : a list containing the names of the output
+
+    types : a list containing the types of the output
+
+    Returns
+    -------
+    pd.DataFrame
+
+    Examples
+    --------
+    >>> coerce_to_dataframe(pd.DataFrame({'a': [1, 2, 3]}), ['b'], [dt.int32])
```

Comment:
can you show the .dtypes after as well (e.g. df = coerce_to....., then show df and df.dtypes)

Reply:
Yes, updated